.circleci/config.yml

```diff
@@ -770,22 +815,32 @@ jobs:
           name: Upload result to codecov.io
           command: bash <(curl -s https://codecov.io/bash) -f $CODECOV_OUTPUT/lcov.info -F unittest;
       - run:
-          name: Grcov html report to S3
+          name: Push Coverage Reports to S3
           command: |
             set -x
-            export AWS_REGION=$AWS_DEFAULT_REGION
-            export GIT_SHORT_REV=$(git rev-parse --short=8 HEAD)
-            export reportdir=${CODECOV_OUTPUT}/grcovhtml;
-            _bucket="ci-artifacts.libra.org/coverage/unit-coverage/$(date +"%Y-%m-%d")-${GIT_SHORT_REV}";
-            aws s3 cp --recursive $reportdir "s3://$_bucket";
-            echo "have to reupload as aws user doesn't have read access";
-            aws s3 cp --recursive $reportdir "s3://ci-artifacts.libra.org/coverage/unit-coverage/latest";
-            _s3_url="https://ci-artifacts.libra.org/coverage/unit-coverage/latest/index.html";
-            echo "Grcov available in s3 ${_s3_url}" >> ${MESSAGE_PAYLOAD_FILE}
+            SUFFIX="$(date +"%Y-%m-%d")-$(git rev-parse --short=8 HEAD)"
+            PREFIX="ci-artifacts.libra.org/coverage";
+            #Push grcov
+            aws s3 cp --recursive ${CODECOV_OUTPUT}/grcovhtml "s3://${PREFIX}/unit-coverage/${SUFFIX}/";
+            aws s3 cp --recursive ${CODECOV_OUTPUT}/grcovhtml "s3://${PREFIX}/unit-coverage/latest/";
+            echo "Grcov available in s3 https://${PREFIX}/unit-coverage/${SUFFIX}/index.html" >> ${MESSAGE_PAYLOAD_FILE}
+            #Push lcov
+            aws s3 cp --recursive ${CODECOV_OUTPUT}/lcovhtml "s3://${PREFIX}/lcov-unit-coverage/${SUFFIX}/";
+            aws s3 cp --recursive ${CODECOV_OUTPUT}/lcovhtml "s3://${PREFIX}/lcov-unit-coverage/latest/";
+            echo "lcov available in s3 https://${PREFIX}/lcov-unit-coverage/${SUFFIX}/index.html" >> ${MESSAGE_PAYLOAD_FILE}
       - send_message:
           payload_file: "${MESSAGE_PAYLOAD_FILE}"
           build_url: "${CIRCLE_BUILD_URL}#tests/containers/${CIRCLE_NODE_INDEX}"
           webhook: "${WEBHOOK_COVERAGE_INFO}"
+      - run:
+          name: Update cache key file
```

Comment:
Will this conflict with the other cache check? Whichever runs first will update the cache key and then the other will not run when it should.

Reply:
No, different cache keys means a different cache will be pulled in.   I could change the file names if you find this confusing.   Of course if might be more confusing the other way different files different cache keys not visible togehter.